cas-server-core/src/main/java/org/jasig/cas/util/DefaultCipherExecutor.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.util;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jose4j.jwe.ContentEncryptionAlgorithmIdentifiers;
+import org.jose4j.jwe.JsonWebEncryption;
+import org.jose4j.jwe.KeyManagementAlgorithmIdentifiers;
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.keys.AesKey;
+import org.jose4j.lang.ByteUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.NotNull;
+import java.security.Key;
+
+/**
+ * The {@link org.jasig.cas.util.DefaultCipherExecutor} is the default
+ * implementation of {@link org.jasig.cas.util.CipherExecutor}. It provides
+ * a facade API to encrypt, sign, and verify values.
+ *
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+public final class DefaultCipherExecutor implements CipherExecutor {
```

Comment:
The `CipherExecutor` needs to be CAS specific, so it can be pulled out or replaced with other impls I suppose. We are using a library for the encryption and signing of course but the wrapper on top of those calls would be better served if it was CAS code. Much like the HttpClient wrapper we have.